### PR TITLE
Move to OpenTracing KeyValue logging

### PR DESCRIPTION
**Pull request by bensigelman**
cc @jmacd and @aliceadelef regarding the treatment of the reportBuffer, especially (which led to some refactoring). I decided to go for an encoding error count rather than internal log records and also changed the way parameters are passed around.

I cleaned up a few misc TODOs along the way, too.

### CODE REVIEW

**Comment by bensigelman** [comment]
(See PR 108 in the OT-Go repo for context)

File: logencoder.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+import (
+	"encoding/json"
+
+	cpb "github.com/lightstep/lightstep-tracer-go/collectorpb"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+const (
+	ellipsis = "…"
```
Comment (aliceadelef) [inline logencoder.go]: ? why "&"

File: thrift_rpc/logencoder.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"fmt"
+
+	"github.com/lightstep/lightstep-tracer-go/lightstep_thrift"
+	"github.com/lightstep/lightstep-tracer-go/thrift_0_9_2/lib/go/thrift"
+	"github.com/opentracing/opentracing-go/log"
+)
+
+const (
+	deprecatedFieldKeyEvent   = "event"
+	deprecatedFieldKeyPayload = "payload"
```
Comment (djspoons) [inline thrift_rpc/logencoder.go]: You can delete the original versions of these (though the "event" one is called "message")

File: recorder.go [hunk 3]
```diff
@@ -28,8 +26,9 @@ import (
 )
 
 const (
-	spansDropped  = "spans.dropped"
-	collectorPath = "/_rpc/v1/reports/binary"
+	spansDropped     = "spans.dropped"
+	logEncoderErrors = "log_encoder.errors"
+	collectorPath    = "/_rpc/v1/reports/binary"
```
Comment (djspoons) [inline recorder.go]: While you're here, `collectorPath` is unused.

Comment (bensigelman) [inline recorder.go]: (not unused AFAICT...)

**Pull request merged**